scargo/config.py

```diff
@@ -91,42 +91,32 @@ class ProjectConfig(BaseModel):
     def target(self) -> "Target":
         return Target.get_target_by_id(self.target_id)
 
+    def get_compiler_warning(self) -> Optional[str]:
+        if (self.cc and not self.target.cc) or (self.cxx and not self.target.cxx):
+            return "Compiler settings are ignored for this target"
+        return None
+
 
 class Target(BaseModel):
     id: str
     family: str
     source_dir: str
-    cc: str
-    cxx: str
+    cc: Optional[str] = None
+    cxx: Optional[str] = None
 
     @classmethod
     def get_target_by_id(cls, target_id: str) -> "Target":
         return TARGETS[target_id]
 
 
-ARM_CC = "arm-none-eabi-gcc"
-ARM_CXX = "arm-none-eabi-g++"
-
 TARGETS = {
     "x86": Target(id="x86", family="x86", source_dir="src", cc="gcc", cxx="g++"),
-    "stm32": Target(
-        id="stm32", family="stm32", source_dir="src", cc=ARM_CC, cxx=ARM_CXX
```

Comment:
Compiler settings for stm32 should work, right? We should leave `arm-none-eabi-*` here.

Reply:
No, these settings are ignored for stm32. They're only used for x86, in `x86.cmake.j2`. So including them for stm32 is equally misleading as for esp32.